metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -583,76 +583,76 @@ void optimize_anchors_in_row_diff(const std::string &graph_fname,
 
     logger->trace("Optimizing anchors");
 
-    std::vector<std::string> filenames;
+    std::vector<sdsl::int_vector_buffer<>> row_reduction;
     for (const auto &p : std::filesystem::directory_iterator(dest_dir)) {
         auto path = p.path();
         if (utils::ends_with(path, row_reduction_extension)) {
             logger->info("Found row reduction vector {}", path);
-            filenames.push_back(path);
+            row_reduction.emplace_back(path, std::ios::in, 1024 * 1024);
+
+            if (row_reduction.back().size() != row_reduction.front().size()) {
+                logger->error("Row reduction vectors have different sizes");
+                exit(1);
+            }
+            if (row_reduction.back().width() != row_reduction.front().width()) {
+                logger->error("Row reduction vectors have different width: {} != {}",
+                              row_reduction.back().width(),
+                              row_reduction.front().width());
+                exit(1);
+            }
         }
     }
 
-    if (!filenames.size()) {
+    if (!row_reduction.size()) {
         logger->error("Didn't find any row reduction vectors in {} to merge", dest_dir);
         exit(1);
     }
 
-    if (filenames.size() > 1u)
-        logger->trace("Merging row reduction vectors");
-
-    while (filenames.size() > 1u) {
-        std::vector<std::string> filenames_new((filenames.size() + 1) / 2);
+    sdsl::bit_vector anchors;
+    uint64_t num_anchors_old;
+    {
```

Comment:
Is this block intentional? It's already delimited by newlines, doesn't seem to bring much in readability

Reply:
It's to automatically remove all temp variables like `ifstream` and `old_anchors` which will not be needed below.

Reply:
Maybe add a comment at the beginning of the block to justify its existence. 

Did you notice any difference when not using the block? I suspect the compiler will destroy those variables for you anyway when they are not needed if it can apply the "as if" rule.

Reply:
It's just for readability.

The function is very large, so I want to avoid creating and keeping variables that are never needed below.
Otherwise, I would have to keep everything in my head, because never know if that variable will or will not be used later.
This thing clearly shows -- these are local temporary objects that are not used later.

Also, I can use that variable name below without messing with those that are created above.